recipes/libexif/all/conanfile.py

```diff
@@ -0,0 +1,80 @@
+from conans import ConanFile, tools, AutoToolsBuildEnvironment
+import os
+
+required_conan_version = ">=1.33.0"
+
+
+class LibexifConan(ConanFile):
+    name = "libexif"
+    url = "https://github.com/conan-io/conan-center-index"
+    homepage = "https://libexif.github.io/"
+    license = "LGPL-2.1"
+    description = "libexif is a library for parsing, editing, and saving EXIF data."
+    topics = ("exif", "metadata", "parse", "edit")
+    settings = "os", "arch", "compiler", "build_type"
+    options = {
+        "shared": [True, False],
+        "fPIC": [True, False],
+    }
+    default_options = {
+        "shared": False,
+        "fPIC": True,
+    }
+    _autotools = None
+
+    @property
+    def _source_subfolder(self):
+        return "source_subfolder"
+
+    @property
+    def _settings_build(self):
+        return getattr(self, "settings_build", self.settings)
+
+    def config_options(self):
+        if self.settings.os == "Windows":
+            del self.options.fPIC
+
+    def configure(self):
+        if self.options.shared:
+            del self.options.fPIC
+        del self.settings.compiler.libcxx
+        del self.settings.compiler.cppstd
+
+    def build_requirements(self):
+        self.build_requires("gettext/0.21")
+        self.build_requires("libtool/2.4.6")
+        if self._settings_build.os == "Windows" and not tools.get_env("CONAN_BASH_PATH"):
+            self.build_requires("msys2/cci.latest")
+
+    def source(self):
+        tools.get(**self.conan_data["sources"][self.version], destination=self._source_subfolder, strip_root=True)
+
+    def _configure_autotools(self):
+        if self._autotools:
+            return self._autotools
+        self._autotools = AutoToolsBuildEnvironment(self, win_bash=self._settings_build.os == "Windows")
+        yes_no = lambda v: "yes" if v else "no"
+        args = [
+            "--enable-shared={}".format(yes_no(self.options.shared)),
+            "--enable-static={}".format(yes_no(not self.options.shared)),
+        ]
+        self._autotools.configure(args=args, configure_dir=self._source_subfolder)
+        return self._autotools
+
+    def build(self):
+        autotools = self._configure_autotools()
```

Comment:
You need to handle Visual Studio specially.
See other cci recipes how this can be done (using a `_build_context` settings `CC/CXX` + `tools.vcvars`.
Though, that is no guarantee to success: https://github.com/libexif/libexif/blob/master/README-Win32.txt

Reply:
Frankly, that stuff is way over my head. But I tried 😄